modules/common/condition/funcs_test.go

```diff
@@ -422,6 +422,45 @@ func TestMirror(t *testing.T) {
 	g.Expect(targetCondition.Message).To(BeIdenticalTo(trueReady.Message))
 }
 
+func TestMirrorReadyTrueAndFalse(t *testing.T) {
+	g := NewWithT(t)
+
+	conditions := Conditions{}
+	// We expect that ReadyCondition True means the overall status of the
+	// resource is ready that condition is mirrored even if there are other
+	// conditions in non True state.
+	conditions.Init(&Conditions{*trueReady, *falseB, *unknownA})
+
+	targetCondition := conditions.Mirror("targetConditon")
+
+	g.Expect(targetCondition.Status).To(BeIdenticalTo(trueReady.Status))
+	g.Expect(targetCondition.Severity).To(BeIdenticalTo(trueReady.Severity))
+	g.Expect(targetCondition.Reason).To(BeIdenticalTo(trueReady.Reason))
+	g.Expect(targetCondition.Message).To(BeIdenticalTo(trueReady.Message))
+}
+
+func TestMirrorInvalidStatus(t *testing.T) {
+	g := NewWithT(t)
+
+	conditions := Conditions{}
+	invalidStatusA := Condition{
+		Type:     "a",
+		Status:   "FooBar",
+		Reason:   "",
+		Severity: SeverityNone,
+		Message:  "",
+	}
+	conditions.Init(&Conditions{invalidStatusA})
+	// NOTE(gibi): we always have a ReadyCondition added automatically so it
```

Comment:
if ReadyConfition is added automatically (upon initialization?) we probably should not give it a maximum weight, when mirroring it as Ready as well?..

Reply:
We only give it a maximum weight if it is ReadyCondition=True.

There is an assumption that ReadyCondition=True means that the overall resource is ready. We give this higher precedence than to the case when ReadyCondition=True + FooCondition=Unknown (or FooCondition=False) as we consider those as either software faults (i.e. inconsistency in ReadyCondition handling) or an edge case of operators wanting to represent an optional and currently non requested feature of the resource with a FooCondition=Unknown status.

Reply:
I agree that this error condition is not easy to trigger. If the caller never explicitly removes ReadyCondition from the conditions list then an condition with invalid status will always be lower prio than the ReadyCondition itself and therefore be ignored. So this is a highly speculative unit test to cover the case that is more clear for the reader of Mirror implementation than the user of the that implementation. Still I'd like to keep this unit test as otherwise we have a panic case in Mirror that is not covered.